lua/entities/gmod_wire_expression2/core/custom/collcallback.lua

```diff
@@ -0,0 +1,190 @@
+E2Lib.RegisterExtension("collcallback", true)
+
+local maxcollclk = CreateConVar( "wire_col_detector_e2_maxcollclk", 100, 0, "Maximum amount of props with collision detection per gate" )
+local collallow = CreateConVar( "wire_col_detector_e2_collallow", 1, 0, "Allow collision detection of other players' stuff" )
+
+local registered_ents = {}
+
+registerCallback("construct",function(self)
+	self.colDatas = {}
+	self.colEnts = {}
+	self.collFilter = true
+	self.collIgnoreConstrained = true
+	self.colProps=0
+end)
+
+hook.Add("EntityRemoved", "Raf2EntityRemoved",function (ent)
+	registered_ents[ent]=nil
+end)
+
+function entitiesCollide(ent,data)
+	for i = 1,#registered_ents[ent],1 do
+		self = registered_ents[ent][i]
+		if not IsValid(self.entity) then
+			table.remove(registered_ents[ent],i)
+			i = i-1
+		else
+			if (self.collFilter and #self.colDatas ~= 0) or (self.collIgnoreConstrained and table.HasValue(constraint.GetAllConstrainedEntities(ent),data.HitEntity)) then return end
```

Comment:
This can be split up into:
````
if self.collFilter and #self.colDatas ~= 0 then return end
if self.collIgnoreConstrained and table.HasValue(constraint.GetAllConstrainedEntities(ent),data.HitEntity)) then return end
````